# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/IonTypes.h
# Commit: 1ab8383d17ee
# Full Hash: 1ab8383d17ee1e5db164339236d64bcb7168b225
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-18 16:09:29
# Regressor Bug: 1727084
# File Overlap Count: 1
# Description:
#   Bug 1727084 - Memory64 - Atomic-rmw baseline operations. r=yury
#   
#   Atomic ADD, SUB, AND, OR, XOR, XCHG, CMPXCHG, as well as WAIT and
#   NOTIFY for the baseline compiler.
#   
# ==============================================================================

diff -r 6089b4360edc -r 1ab8383d17ee js/src/jit/IonTypes.h
--- a/js/src/jit/IonTypes.h	Mon Oct 18 07:31:55 2021 +0000
+++ b/js/src/jit/IonTypes.h	Mon Oct 18 07:31:56 2021 +0000
@@ -915,17 +915,25 @@
   Args_Int32_GeneralInt64Int32Int32Int32 = detail::MakeABIFunctionType(
       ArgType_Int32, {ArgType_General, ArgType_Int64, ArgType_Int32,
                       ArgType_Int32, ArgType_Int32}),
+  Args_Int32_GeneralInt64Int32 = detail::MakeABIFunctionType(
+      ArgType_Int32, {ArgType_General, ArgType_Int64, ArgType_Int32}),
+  Args_Int32_GeneralInt64Int32Int64 = detail::MakeABIFunctionType(
+      ArgType_Int32,
+      {ArgType_General, ArgType_Int64, ArgType_Int32, ArgType_Int64}),
   Args_Int32_GeneralInt64Int32Int64General = detail::MakeABIFunctionType(
       ArgType_Int32, {ArgType_General, ArgType_Int64, ArgType_Int32,
                       ArgType_Int64, ArgType_General}),
+  Args_Int32_GeneralInt64Int64Int64 = detail::MakeABIFunctionType(
+      ArgType_Int32,
+      {ArgType_General, ArgType_Int64, ArgType_Int64, ArgType_Int64}),
   Args_Int32_GeneralInt64Int64Int64General = detail::MakeABIFunctionType(
       ArgType_Int32, {ArgType_General, ArgType_Int64, ArgType_Int64,
                       ArgType_Int64, ArgType_General}),
 
   // Functions that return Int64 are tricky because SpiderMonkey's ReturnRegI64
   // does not match the ABI int64 return register on x86.  Wasm only!
-  Args_Int64_General = detail::MakeABIFunctionType(
-      ArgType_Int64, {ArgType_General}),
+  Args_Int64_General =
+      detail::MakeABIFunctionType(ArgType_Int64, {ArgType_General}),
   Args_Int64_GeneralInt64 = detail::MakeABIFunctionType(
       ArgType_Int64, {ArgType_General, ArgType_Int64}),
 